### DISCUSSION TITLE
WM5 feedback and new strategies

### DISCUSSION THREAD

**Discussion by Dedekind561** [CodeYourFuture/curriculum#313] [General]
# WM5 feedback

Hello folks, 
I spent the day with West Midlands last Saturday and it was an extremely productive session. I gained lots of useful feedback, which I’ll summarise here ( but it will probably also be worthwhile covering some of this ground at the next syllabus meeting too ).

## In its current form, prep isn’t effective

We decided not to re-record videos again for everything in JS1. However, trainees have made it clear it's very difficult to digest new curriculum content as text ( even though there are exercises and activities ). I agree with this view and will outline strategies to address this problem below.

## Workshops should be tightly coupled to the current week’s content

On Saturday trainees participated in an [evaluation-style workshop](https://github.com/CodeYourFuture/CYF-Workshops/tree/main/js1-wk1-eval) for concepts and topics covered in JS1 week 1, which served as a kind of formative assessment for the week’s topics. It was the pretty much unanimous view of everyone in WM5 that this kind of workshop was more useful than the one originally scheduled in the day plan ( based on the Git CLI ). Going forward, workshops should evaluate and complement the current week's content instead of trying to introduce brand-new stuff. Very useful feedback.

## JS1 exercises need more direction
Due to the lack of Module-JS1 pulls I was starting to fear that the exercises were way off in terms of difficulty. To be sure, they are more challenging but I think with some modifications we can make them more accessible. Some of the new questions require trainees to appeal to docs to make sense of unfamiliar code. We need to be more specific about which docs to use ( to begin with at least )  so trainees are not spending hours sifting through documentation at the expense of developing their mental models of JS1. Of course, searching through docs is critical, but I think we're assuming they're way more comfortable looking up stuff independently than they are at this stage in the course. I've fallen foul of this. But again, this is something we can improve very quickly.

# 🧭 Strategies

## Moving from writing to pedagogy and classroom practices
The curriculum is our pedagogic software. But I've spent a lot of time writing it but now I need to dedicate more of my time to seeing how it runs in classroom settings. Working with trainees like West Midlands provided so much useful feedback that I can relay back to Global Syllabus so we can continue interesting

## More time on teaching
At the moment, I'm writing but I'm going to switch more of my time to teaching. I plan on teaching the prep content to small groups of trainees ( four or five ) to create a supplementary set of resources that trainees can use alongside the written prep material. I aim to do it on Tuesdays. These recorded sessions will maximise audience participation, be capped at 1 hour, be done in a single take and will continually refer back to the prep source on the curriculum website. Of course, I can't record these workshops for every new class. These recordings can be reused by other classes but I think we can train volunteers to run similar workshops ( especially when they become more familiar with the content ). Fortunately, delivering these workshops won't require planning: the prep content is already a plan! 💪

## Offloading rewrite work

I briefly discussed this in the last syllabus meeting, but I will be offloading curriculum rewrite work in the coming weeks. I will spend the next week and a bit handing over work and ensuring it is as clear as possible which pieces of work are outstanding. But to summarise here, I will finish and complete:

- JS2 Week 3 prep
- JS3 Week 1 prep 
- JS3 Week 2 prep 
- JS3 Week 3 prep

( I think JS2 week 4 and JS3 week 4 should be project/hackathon focused )

The core of the work I will offload will be coursework exercises, day plans and workshops for Saturdays.

## Feedback loop
As I spend more time in class and teaching trainees online I will continue to share valuable insights that we can act upon :)

**Comment by SallyMcGrath** [comment]
Can you open a ticket for the JS1 workshop change?
